compiler/src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala

```diff
@@ -1542,8 +1542,6 @@ class TreeUnpickler(reader: TastyReader,
               // as the reduction of the match type definition!
               //
               // We also override the type, as that's what Typer does.
-              // The difference here is that a match type that reduces to a non-match type
-              // makes the TypeRef for that definition will have a TypeAlias info instead of a MatchAlias.
```

Comment:
Indeed, I agree this shouldn't be impacted by the changes in #19871.
But I'm now not sure about which cases get reduced more from the Unpickler than when from the Typer ?

Reply:
Nothing more.  The point is the have the same definition, whether from source code or from tasty, have the same info, so that differences in MatchAlias and TypeAlias don't become a factor.

Reply:
So assuming the match type reduces to non match type, won't they all have a TypeAlias now ?

Reply:
Correct, but only because I added `overwriteType` here.  The deleted comment is part of (attempting to!) explain that and shouldn't have been removed.

Reply:
Okay, so I agree we need `tpt.overwriteType(tpt.tpe.normalized)` and that it is because `that's what Typer does`.

Still, it was my understanding that the Typer and Unpickler now use AliasingBounds in a consistent way, so I don't understand how there is a `difference here` for `the TypeRef for that definition`.

Reply:
By "The difference here" I meant "The reason that is important to do is because..."